### DISCUSSION TITLE
RFEM 6 API: get_all_selected_objects returns "UNKNOWN" type

### DISCUSSION THREAD

**Discussion by sonkaste1** [Dlubal-Software/Dlubal_CSharp_Client#42] [Q&A] [answered]
Hi there,

we have encountered unexpected response when invoking _get_all_selected_objects_ in RFEM 6.02.0073. 

Here is a dump of the response:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/"
    xmlns:soap-enc="http://schemas.xmlsoap.org/soap/encoding/"
    xmlns:xsd="http://www.w3.org/2001/XMLSchema"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <soap:Body>
        <n1:get_all_selected_objectsResponse xmlns:n1="http://www.dlubal.com/rfem.xsd">
            <value>
                omitting unrelated elements...
                <object_location>
                    <type>UNKNOWN</type>
                    <no>1</no>
                </object_location>
            </value>
        </n1:get_all_selected_objectsResponse>
    </soap:Body>
</soap:Envelope>
```

"UNKNOWN" type was not expected because it is not in the _object_types_ enum. This causes an exception during deserialization on our side.

Is UNKNOWN a new enum? Or it is an issue in API response?

Thank you for answering. :-)
Štěpán

**Comment by Rebecca-Coding21** [comment, marked as answer]
Hi @sonkaste1,
unfortunately I could not reproduce this behaviour in the current customer version (6.03.0001). UNKNOWN is no new enum and therefore shouldn't be returned. Which object type did you select in your model? For further analysis of the problem it would be good if you could provide a detailed description and the related RFEM model.

**Reply by sonkaste1** [reply]
Hi, I will check the behaviour further with colleagues and come back to you ASAP. Thx.

**Reply by sonkaste1** [reply]
Hi @Rebecca-Coding21 , 
"unknown" is response when RFEM steel Connection is selected for API response.
(version 6.03.0005)

![image](https://github.com/Dlubal-Software/Dlubal_CSharp_Client/assets/87471266/cdfcf610-2998-4fd9-afc5-3592c6d03578)

**Reply by Rebecca-Coding21** [reply]
Hi @sonkaste1,
there is no object_type for steel connections at the moment. Therefore they can't be accessed by webservice, respectively return type UNKNOWN. However, it is planned to implement webservice functions for steel joints in the future.

**Reply by sonkaste1** [reply]
Hi @Rebecca-Coding21 ,
thank you, It is clear now.

**Reply by sonkaste1** [reply]
@Rebecca-Coding21 the first step might be to not throw an exception during deserialization for the "UNKNOWN" type.
